### DISCUSSION TITLE
How to show nativemenu only on MacOS and menubar only on Windows with cross-platform app?

### DISCUSSION THREAD

**Discussion by Deracination** [AvaloniaUI/Avalonia#11470] [General]
It's unclear to me how to customize the UI so that MacOS can use the NativeMenu and have no in-window menus (per MacOS standards) and for Windows show the usual menubar across the top of the window.

NativeMenu doesn't support IsVisible so hide/show via attributes won't work?

Is there a conditional compilation ifdef or some such which can be used in the XAML? How should this be done?

**Comment by kekekeks** [comment]
1) define NativeMenu for the window
2) add NativeMenuBar control to the place where you would like to show the menu on Windows, it will automatically pick items from window's NativeMenu menu on platforms that don't support it

**Reply by Deracination** [reply]
Thanks, perfect.

```
 <NativeMenuItem Header="Activities">
   <NativeMenuItem.Menu>
     <NativeMenu>
       <NativeMenuItem Header="Create New Activity" Command="{Binding AddActivityCommand}"/>
      </NativeMenu>
  </NativeMenuItem.Menu>
</NativeMenuItem>

 <DockPanel>
   <NativeMenuBar DockPanel.Dock="Top" x:Name="MenuBar" />
etc, etc
```
